src/test/java/com/wikia/webdriver/testcases/desktop/categoriestests/DynamicCategoryTests.java

```diff
@@ -0,0 +1,109 @@
+    new ArticlePageObject().open("Category:Wikia_tests");
+
+    DynamicCategoryPage categoryPage = new DynamicCategoryPage();
+    CategoryLayoutSelector layoutSelector = categoryPage.getLayoutSelector();
+    Assertion.assertTrue(layoutSelector.isCategoryExhibitionActive());
+
+  }
+
+  @Execute(asUser = User.USER)
+  public void dynamicCategoryLoggedInUserCanChangeLayout() {
+    new ArticlePageObject().open("Category:Wikia_tests");
+
+    DynamicCategoryPage categoryPage = new DynamicCategoryPage();
+    CategoryLayoutSelector layoutSelector = categoryPage.getLayoutSelector();
+    Assertion.assertTrue(layoutSelector.isVisible(), "Layout selectors are displayed for anon");
+    String pageUrl = categoryPage.getCurrentUrl();
+    layoutSelector.switchToDynamicCategoriesView();
+    Assertion.assertStringContains(categoryPage.getCurrentUrl(), pageUrl);
+  }
+
+  @Execute(asUser = User.USER_2)
+  public void pagination200elementsNoNextButton() {
+    new ArticlePageObject().open("Category:200elements");
+
+    DynamicCategoryPage categoryPage = new DynamicCategoryPage();
+    CategoryLayoutSelector layoutSelector = categoryPage.getLayoutSelector();
+    Assertion.assertTrue(layoutSelector.isDynamicViewActive());
+    PaginationControls paginationControls = categoryPage.getPaginationControls();
+    Assertion.assertFalse(paginationControls.isNextButtonVisible());
+  }
+
+  @Execute(asUser = User.USER_2)
+  public void pagination201elementsNextButtonVisible() {
+    new ArticlePageObject().open("Category:201elements");
+
+    DynamicCategoryPage categoryPage = new DynamicCategoryPage();
+    CategoryLayoutSelector layoutSelector = categoryPage.getLayoutSelector();
+    Assertion.assertEquals(layoutSelector.isDynamicViewActive(), true, "Dynamic view is not active");
+    PaginationControls paginationControls = categoryPage.getPaginationControls();
+    paginationControls.clickNextButton();
+    Assertion.assertEquals(categoryPage.getMembers().size(), 1, "Invalid number of members on category page");
+  }
+
+  @Execute(asUser = User.USER_2)
+  public void pagination401elementsNextButtonVisible() {
+    new ArticlePageObject().open("Category:401elements");
+
+    DynamicCategoryPage categoryPage = new DynamicCategoryPage();
+    CategoryLayoutSelector layoutSelector = categoryPage.getLayoutSelector();
+    Assertion.assertEquals(layoutSelector.isDynamicViewActive(), true, "Dynamic view is not active");
+    PaginationControls paginationControls = categoryPage.getPaginationControls();
+    Assertion.assertTrue(paginationControls.isNextButtonVisible(), "Next button is not visible");
+    Assertion.assertTrue(paginationControls.isLastButtonVisible(), "Last button is not visible");
+    paginationControls.clickLastButton();
+
+    List<WebElement> memberList = categoryPage.getMembers();
+
+    Assertion.assertEquals(categoryPage.getMembers().size(), 1);
+    Assertion.assertTrue(memberList.stream().anyMatch(e->e.getText().equals("Article99")));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the "Article99" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1132)